src/Common/Core/Impl/Services/ServiceManager.cs

```diff
@@ -54,7 +54,19 @@ public class ServiceManager : IServiceManager {
         public virtual IServiceManager AddService(Type type) {
             _disposeToken.ThrowIfDisposed();
 
-            var lazy = new Lazy<object>(() => Activator.CreateInstance(type));
+            object parameter = null;
```

Comment:
There is an extension method that uses compiled expression tree: `ServiceManagerExtensions.AddService`. I'm actually not sure if we need `ServiceManager.AddService(Type type)` at all